renku/core/dataset/dataset_add.py

```diff
@@ -155,6 +159,21 @@ def _download_files(
     **kwargs,
 ) -> List["DatasetAddMetadata"]:
     """Process file URLs for adding to a dataset."""
+    if dataset.storage and dataset.storage.lower().startswith("s3://"):
+        if len(urls) == 0:
+            raise errors.ParameterError("No URL is specified")
+        if len(urls) > 1:
+            raise errors.ParameterError("Can only add 1 uri with S3 datasets.")
+        if not urls[0].lower().startswith("s3://"):
```

Comment:
This check can be moved to the for loop if we allow more than one URL

Reply:
We should generalize this check to allow other providers. Maybe by adding a function to the dataset class.

Reply:
I removed/changed it so that it is more general.